src/utilities/filetypes/EpwFile.cpp

```diff
@@ -2941,7 +2956,11 @@ void EpwDesignCondition::setHeatingDryBulb99pt6(double heatingDryBulb99pt6) {
 bool EpwDesignCondition::setHeatingDryBulb99(const std::string& heatingDryBulb99) {
   bool ok;
   double value = stringToDouble(heatingDryBulb99, &ok);
-  setHeatingDryBulb99(value);
+  if (!ok) {
+    m_heatingDryBulb99 = boost::none;
+  } else {
+    setHeatingDryBulb99(value);
+  }
   return ok;
```

Comment:
Not in the scope of this PR, but I dislike this. `stringToDouble` should be `boost::optional<double> stringToDouble(const std::string& str)`.

Also a lot of fields in `EpwDataPoint` internally store a `std::string` representation of a number, use magic string numbers for unknown values (constants would be better, whether that's a string or an actual number), which I also don't like.

https://github.com/NREL/OpenStudio/blob/cae94352f7d5ce2c44602dbda113fcb7bf085ab8/src/utilities/filetypes/EpwFile.cpp#L1718-L1744

But this isn't something we need to address now.